cli/pcluster/easyconfig.py

```diff
@@ -233,11 +250,101 @@ def configure(args):  # noqa: C901 FIXME!!!
         if e.errno != errno.EEXIST:
             raise  # can safely ignore EEXISTS for this purpose...
 
+    if not _is_config_valid(args, config):
+        sys.exit(1)
+
+    # If we are here, than the file it's correct and we can override it.
     # Write configuration to disk
     open(config_file, "a").close()
     os.chmod(config_file, stat.S_IRUSR | stat.S_IWUSR)
     with open(config_file, "w") as cf:
         config.write(cf)
 
+
+def _is_config_valid(args, config):
+    """
+    Validate the configuration of the pcluster configure.
+
+    :param args: the arguments passed with the command line
+    :param config: the configParser
+    :return True if the configuration is valid, false otherwise
+    """
+    # We create a temp_file_path to validate before overriding the original config
+    temp_file_path = os.path.join(tempfile.gettempdir(), "temp_config")
+    temp_args = copy.copy(args)  # Defensive copy is needed because we change config_file
+
+    temp_args.config_file = temp_file_path
+    with open(temp_file_path, "w+") as cf:
+        config.write(cf)
     # Verify the configuration
-    cfnconfig.ParallelClusterConfig(args)
+    is_valid = True
+    try:
+        cfnconfig.ParallelClusterConfig(temp_args)
+    except SystemExit:
+        is_valid = False
+    finally:
+        os.remove(temp_file_path)
+        return is_valid
+
+
+def get_config_parameter(config, section, parameter_name, default_value):
+    """
+    Get the parameter if present in the configuration otherwise returns default value.
+
+    :param config the configuration parser
+    :param section the name of the section
+    :param parameter_name: the name of the parameter
+    :param default_value: the default string to propose the user
+    :return:
+    """
+    return config.get(section, parameter_name) if config.has_option(section, parameter_name) else default_value
+
+
+def scheduler_handler(config, cluster_label, scheduler):
```

Comment:
Dropping this comment here although it is more generic: I would decouple as much as possible the logic that prompts questions to the customers from the logic that maps the read values to the parallelcluster representation.

Also try to understand how easy it is, after the refactoring of the config parser and validator, to reuse the config definitions and default values from that logic. We would like to avoid as much as possible to have to modify multiple parts of the code when we change default values or allowed values for a given field.